applications/server/controllers/synchronization.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+import json
+
+def index():
+    """
+    example action using the internationalization operator T and flash
+    rendered by views/default/index.html or views/generic.html
+
+    if you need a simple wiki simply replace the two lines below with:
+    return auth.wiki()
+    """
+    response.flash = T("Hello World")
+    return dict(message=T('Welcome to web2py!'))
+
+
+@request.restful()
+def recieve_update():
+    def GET(*args, **vars):
+        return response.json(['foo', {'bar': ('baz', None, 1.0, 2)}])
+
+    def POST(*args, **vars):
+        # get the log data from the request and validate
+        request_body = request.body.read()
+        if (not request_body):
+            raise HTTP(400, "no data was included")
+
+        # get json and validate structure
+        try:
+            log_data = json.loads(request_body)
+        except:
+            raise HTTP(400, "data was not json-formatted")
+        if (not isinstance(log_data, dict)):
+            raise HTTP(400, "data was not properly formatted")
+        if (not "logs" in log_data and not isinstance(log_data.get('logs'), list)):
+            raise HTTP(400, "data needs to have list of log messages as 'logs'")
+        if (not "device_id" in log_data and not isinstance(log_data.get('device_id'), str)):
+            raise HTTP(400, "data needs to have string device id as 'device_id'")
+
+        # get information from document
```

Comment:
So this is purely some test code? 
Also, I don't know that you need to do all this validation.  Another idea is to just use those fields. 
If you run aground, you will generate an error 500, and the logs will contain the information for you to see. 
The device is supposed to know how to talk the correct protocol after all.